robocup_ws/src/Planner/src/player_controller.py

```diff
@@ -282,5 +421,62 @@ def get_action(self, my_pos_efcs:Position):
             self.current_goal += 1
             self.current_goal %= 6
         l_rpm, r_rpm, = simple_go_to_action(my_pos_efcs, goal_pos)
-        print(f"({l_rpm}, {r_rpm})")
-        return PlayerCommand(l_rpm, r_rpm, 0)
\ No newline at end of file
+        return PlayerCommand(l_rpm, r_rpm, 0)
+
+    def distance_judge(self, game_info: list, init_distance = None) -> [bool, int, int]:
+        team = game_info[0]
+        enemy = game_info[1]
+        player_id = game_info[2]
+        mode = game_info[3]
+        enemy_d = []
+        if(init_distance is not None):
+            distance = init_distance
+        elif(mode == "ATTACK"):
+            distance = 0.3
+        elif(mode == "DEFEND"):
+            distance = 0.3
+
+        my_pos = team.players_positions_efcs[player_id]
+        team_pos = team.players_positions_efcs
+        enemy_pos = enemy.players_positions_wcs
+
+        for j in range(5):
+            enemy_distance = np.hypot(enemy_pos[j].x - my_pos.x, enemy_pos[j].y - my_pos.y)
+            if enemy_distance <= distance:
+                return True, 1, j
+
+        for i in range(5):
+            team_distance = np.hypot(team_pos[i].x - my_pos.x, team_pos[i].y - my_pos.y)
+            if (team_distance <= distance) and (team_distance != 0):
+                return True, 0, i
+
+        return False, -1, -1
+
+    def avoid_obstacle(self, game_info: list, team_id, obstacle_player_id) -> PlayerCommand:
+        team = game_info[0]
+        obstacle_team = game_info[team_id]
+        player_id = game_info[2]
+        my_pos = team.players_positions_efcs[player_id]
+        safe_pos = obstacle_team.players_positions_wcs[obstacle_player_id]
+
+        lv, rv = self.go_around_the_point(my_pos, safe_pos, 0.3)
+        return PlayerCommand(lv, rv, 0)
+
+    def go_around_the_point(self, robot_state: Position, go_around_point: Position, radius, direction=1, R_vel=3):
```

Comment:
this is repeated - same in the line 348

Reply:
@Yifan-Xie What do I do with this? ITs for obstacle avoidance